conu/apidefs/container.py

```diff
@@ -59,11 +62,32 @@ def http_request(self, path="/", method="GET", host=None, port=None, json=False,
         """
         host = host or self.get_IPv4s()[0]
         port = port or self.get_ports()[0]
-        url = urlunsplit(
-            ("http", "%s:%s" % (host, port), path, "", "")
-        )
+        url = get_url(host=host, port=port, path=path)
+
         return self.http_session.request(method, url, json=json, data=data)
 
+    @contextmanager
+    def http_client(self, host=None, port=None):
+        """
+        allow requests in context -- e.g.:
+
+        .. code-block:: none
```

Comment:
I think `s/none/python/` to get the correct syntax highlighting.

Reply:
Done.